mbuild/formats/gomc_conf_writer.py

```diff
@@ -0,0 +1,4065 @@
+                                     + output_freq_variables_List + histogram_output_variables_List \
+                                     + output_data_variables_List + free_energy_variables_List \
+                                     + Std_MC_moves_variables_List + MEMC_MC_moves_variables_List
+
+
+    else:
+        print('ERROR: The ensemble_type selected for the _get_possible_ensemble_input_variables function is not valid.')
+        valid_input_variables_List = None
+
+
+    return  valid_input_variables_List
+
+
+
+
+def _get_required_ensemble_files(ensemble_type):
+    """
+    Provides list of the possible optional input variables based on the ensemble type
+
+    Parameters
+    ----------
+    ensemble_type = str; valid options are 'NVT', 'NPT', 'GEMC_NVT', 'GEMC_NPT', 'GCMC'
+
+    Outputs
+    ---------
+    required_ensemble_files_List = list; a list of the required ensemble input variables
+
+    """
+    if ensemble_type in ['NVT', 'NPT']:
+
+        simulation_settings = ["charmm_object", "ensemble_type", "RunSteps", "Temperature"]
+
+        required_ensemble_files_List =  simulation_settings
+
+    elif ensemble_type in ['GEMC_NVT', 'GEMC_NPT', 'GCMC']:
+
+        simulation_settings = ["charmm_object", "ensemble_type", "RunSteps", "Temperature"]
+
+        required_ensemble_files_List =   simulation_settings
+
+
+    else:
+        print('ERROR: The ensemble_type selected for the _get_required_ensemble_files function is not valid.')
+        required_ensemble_files_List = None
+
+    return required_ensemble_files_List
+
+
+
+
+class GOMCControl():
+    def __init__(self, charmm_object, ensemble_type, RunSteps, Temperature, input_variables_dict = None
+                 ):
+
+        """
+
+        Construct the GOMC configuration input file with the defaults,
+        or adding additional data in the input_variable section
+        Default setting for the GOMC configuraion files are based upon the
+        a educated guess which should result in reasonable sampling for a
+        given ensemble/simulation type. However, there is no guarantee that
+        the default setting will provide the best or adequate sampling for
+        the selected system. The user has the option to modify the
+        configuration/contorl files based on the simulation specifics or in to
+        optimize the system beyond the standard settings.  These override
+        options are available via the keyword arguments in input_variable_dict.
+        Parameters
+        ----------
+        charmm_object :  Charmm object, which by definition has been parameterized ' \
+                      + 'from the selected force field.',
+        ensemble_typ : str, only accepts 'NVT', 'NPT', 'GEMC_NPT', 'GCMC-NVT', 'GCMC'
+        RunSteps : int; must be an integer greater than zero.
+            Sets the total number of simulation steps.
+        Temperature : Temperature of system in Kelvin (K)
```

Comment:
conform to numpydoc style, needs data type, description on separate line, no blank lines between each parameter.

Reply:
fixed